src/main/java/io/zksync/protocol/provider/DefaultEthereumProvider.java

```diff
@@ -60,23 +63,41 @@ public CompletableFuture<TransactionReceipt> transfer(Token token, BigInteger am
         }
     }
 
-    public CompletableFuture<BigInteger> getDepositBaseCost(@Nullable BigInteger gasPrice) {
+    @Override
+    public CompletableFuture<BigInteger> getBaseCost(BigInteger gasLimit, BigInteger gasPerPubdataByte, @Nullable BigInteger gasPrice) {
         if (gasPrice != null) {
-            return contract.depositBaseCost(gasPrice, PriorityQueueType.Deque.getValue(), PriorityOpTree.Full.getValue()).sendAsync();
+            return contract.l2TransactionBaseCost(gasPrice, gasLimit, gasPerPubdataByte).sendAsync();
         } else {
             return getGasPrice()
-                    .thenCompose(g -> contract.depositBaseCost(g, PriorityQueueType.Deque.getValue(), PriorityOpTree.Full.getValue()).sendAsync());
+                    .thenCompose(g -> contract.l2TransactionBaseCost(g, gasLimit, gasPerPubdataByte).sendAsync());
         }
     }
 
     @Override
-    public CompletableFuture<TransactionReceipt> deposit(Token token, BigInteger amount, String userAddress) {
-        BigInteger tips = BigInteger.ZERO;
-        if (token.isETH()) {
-            return l1EthBridge.deposit(userAddress, Address.DEFAULT.getValue(), amount, amount.add(tips)).sendAsync();
-        } else {
-            return l1ERC20Bridge.deposit(userAddress, token.getL1Address(), amount, tips).sendAsync();
-        }
+    public CompletableFuture<TransactionReceipt> requestExecute(String contractAddress, BigInteger l2Value, byte[] calldata, BigInteger gasLimit, @Nullable byte[][] factoryDeps, @Nullable BigInteger operatorTips, @Nullable BigInteger gasPrice, String refundRecipient) {
+        return CompletableFuture.supplyAsync(() -> {
+            BigInteger gasPriceValue = gasPrice == null ? getGasPrice().join() : gasPrice;
+            List<byte[]> factoryDepsList = factoryDeps == null ? Collections.emptyList() : Arrays.asList(factoryDeps);
+            BigInteger operatorTipsValue = operatorTips == null ? BigInteger.ZERO : operatorTips;
+            BigInteger baseCost = getBaseCost(gasLimit, L1_TO_L2_GAS_PER_PUBDATA, gasPriceValue).join();
+            BigInteger totalValue = l2Value.add(baseCost).add(operatorTipsValue);
+            return contract.requestL2Transaction(contractAddress, l2Value, calldata, gasLimit, L1_TO_L2_GAS_PER_PUBDATA, factoryDepsList, refundRecipient, totalValue).sendAsync().join();
+        });
+    }
+
+    @Override
+    public CompletableFuture<TransactionReceipt> deposit(Token token, BigInteger amount, BigInteger operatorTips, String userAddress) {
+        return CompletableFuture.supplyAsync(() -> {
+            BigInteger baseCost = BigInteger.ZERO;
```

Comment:
Nit: `baseCost` can be declared in else statement, as it's not used if case when `token.isETH()` is true.